src/main/java/io/nats/client/api/ConsumerConfiguration.java

```diff
@@ -1027,9 +1072,25 @@ protected static Duration getOrUnset(Duration val)
         return val == null ? DURATION_UNSET : val;
     }
 
-    private static Long normalize(Long l)
-    {
-        return l == null ? null : l <= LONG_UNSET ? LONG_UNSET : l;
+    private static Integer normalizeToInt(Number proposed) {
+        return normalizeToInt(proposed, LONG_UNSET + 1);
+    }
+
+    private static Integer normalizeToInt(Number proposed, long min) {
+        if (proposed == null) {
+            return null;
+        }
+
+        long l = proposed.longValue();
+        if (l < min) {
+            return INTEGER_UNSET;
+        }
+
+        if (l > Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
+        }
```

Comment:
A value for these integer fields anywhere near Integer MAX_VALUE would not make sense, so instead of an error if the proposed long value was beyond that, I just assumed the user was trying to set some max value and handled it instead of erroring.